images/charmed-spark-gpu/rockcraft.yaml

```diff
@@ -0,0 +1,239 @@
+        echo "$ACTUAL_SHA"
+        echo "$EXPECTED_SHA"
+        [ $ACTUAL_SHA != $EXPECTED_SHA ] && echo "ERROR"  && exit 1
+      done
+      echo "All dependecies downloaded!"
+    stage:
+      - opt/spark/jars
+
+  spark8t:
+    plugin: nil
+    after: [ dependencies ]
+    build-packages:
+      - wget
+      - ssl-cert
+      - git
+    overlay-packages:
+      - python3-pip
+    overlay-script: |
+      mkdir -p $CRAFT_PART_INSTALL/opt/spark8t/python/dist
+      pip install --target=${CRAFT_PART_INSTALL}/opt/spark8t/python/dist  https://github.com/canonical/spark-k8s-toolkit-py/releases/download/v0.0.7/spark8t-0.0.7-py3-none-any.whl
+      rm usr/bin/pip*
+    stage:
+      - opt/spark8t/python/dist
+
+  kubectl:
+    plugin: nil
+    build-packages:
+      - wget
+    overlay-script: |
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/bin
+      cd $CRAFT_PART_INSTALL/usr/local/bin
+      KUBECTL_VERSION=$(wget -qO- https://dl.k8s.io/release/stable.txt)
+      wget -q "https://dl.k8s.io/release/$KUBECTL_VERSION/bin/linux/amd64/kubectl"
+      wget -q  "https://dl.k8s.io/$KUBECTL_VERSION/bin/linux/amd64/kubectl.sha256"
+      echo "$(cat kubectl.sha256)  kubectl" | sha256sum --check
+      if  [[ $? -ne 0 ]]
+        then
+          echo "DOWNLOAD ERROR: kubectl could not be downloaded properly! Exiting...." >&2
+          exit 1
+      fi
+      chmod +x kubectl
+      cp -r $CRAFT_PART_INSTALL/usr/ $CRAFT_PART_BUILD
+    stage:
+      - usr/local/bin
+
+  charmed-spark:
+    plugin: dump
+    after: [ spark8t, spark ]
+    source: .
+    organize:
+      conf/spark-defaults.conf: etc/spark8t/conf/spark-defaults.conf
+      bin/sparkd.sh: opt/pebble/sparkd.sh
+      bin/history-server.sh: opt/pebble/history-server.sh
+      bin/spark-client.pyspark: opt/spark-client/python/bin/spark-client.pyspark
+      bin/spark-client.spark-sql: opt/spark-client/python/bin/spark-client.spark-sql
+      bin/spark-client.service-account-registry: opt/spark-client/python/bin/spark-client.service-account-registry
+      bin/spark-client.spark-shell: opt/spark-client/python/bin/spark-client.spark-shell
+      bin/spark-client.spark-submit: opt/spark-client/python/bin/spark-client.spark-submit
+      bin/getGpusResources.sh: opt/getGpusResources.sh
+    stage:
+      - etc/spark8t/conf/
+      - opt/pebble/sparkd.sh
+      - opt/pebble/history-server.sh
+      - opt/spark-client/python/bin/spark-client.pyspark
+      - opt/spark-client/python/bin/spark-client.spark-sql
+      - opt/spark-client/python/bin/spark-client.service-account-registry
+      - opt/spark-client/python/bin/spark-client.spark-shell
+      - opt/spark-client/python/bin/spark-client.spark-submit
+      - opt/getGpusResources.sh
+
+  user-setup:
+    plugin: nil
+    after: [ charmed-spark ]
+    overlay-packages:
+      - tini
+      - libc6
+      - libpam-modules
+      - krb5-user
+      - libnss3
+      - procps
+      - openjdk-8-jre
+      - openjdk-8-jdk
```

Comment:
The rock image currently uses `openjdk-17-jre-headless`. Maybe we can use the same version with this image as well?

Reply:
We need to investigate compatibility with Java 17 in a future ticket